src/HashMap.mo

```diff
@@ -167,21 +322,50 @@ module {
 
   };
 
-  /// clone cannot be an efficient object method,
-  /// ...but is still useful in tests, and beyond.
+  /// Returns a copy of `map`, initializing the copy with the provided equality
+  /// and hash functions.
+  ///
+  /// Example:
+  /// ```motoko include=initialize
+  /// map.put("key1", 1);
+  /// map.put("key2", 2);
+  /// map.put("key3", 3);
+  ///
+  /// let map2 = HashMap.clone(map, Text.equal, Text.hash);
+  /// map2.get("key1") // ?1
+  /// ```
+  ///
+  /// Expected Runtime: O(size), Worst Case Runtime: O(size * size)
+  ///
+  /// Expected Space: O(size), Worst Case Space: O(size * size)
```

Comment:
Detail (also applies to `map` and `mapFilter`): I am not sure, whether the worst case space is `O(size * size)`, not `O(size)`. If all entries would collide with the hash function, the sum of the collision lists is `O(size)` plus the hash table array that is also `O(size)`. Or probably, it is the garbage amount that contributes to `O(size * size)` (this could be indeed the case).

Reply:
Ah yes you're right. The worst case space should be `O(size)`. I will update before merging. Thank you!